src/modules/flow/http-client/http-client.c

```diff
@@ -60,6 +62,21 @@ struct http_client_node_type {
     int (*process_data)(struct sol_flow_node *node, struct sol_http_response *response);
 };
 
+static int
+machine_id_header_add(struct sol_http_param *params, char *id)
```

Comment:
Well, the `sol_platform_get_machine_id()` call is strange, it should cache the value and return a pointer to the internal buffer.

would you change that as well, then your call will be simplified as you can keep the reference to that buffer.